### DISCUSSION TITLE
Switch to automatic Android plugin loading. Plugin not implemented.

### DISCUSSION THREAD

**Discussion by johnborges** [ionic-team/capacitor#4984] [Q&A]
I'm trying to better understand how automatic plugin loading should work. I've added the android project generated from `@capacitor/android@3.2.0` for an app that already is working fine for iOS. 

`npx cap add android`

My app already has a number of plugins installed since the iOS platform was added first. My `package.json` and `MainActivity.java` looks like this:

```json
{
  "name": "example",
  "private": true,
  "version": "0.0.1",
  "description": "example",
  "license": "MIT",
  "files": [
    "dist/"
  ],
  "scripts": {
    "build": "stencil build --dev",
    "start": "stencil build --dev --watch --serve",
    "test": "stencil test --spec --e2e",
    "test.watch": "stencil test --spec --e2e --watch",
    "generate": "stencil generate"
  },
  "devDependencies": {
    "@ionic/core": "^5.3.2",
    "@stencil/core": "^2.0.1"
  },
  "dependencies": {
    "mycustom-plugin": "^0.5.0",
    "@capacitor/android": "^3.2.0",
    "@capacitor/app": "^1.0.0",
    "@capacitor/cli": "^3.2.0",
    "@capacitor/core": "^3.2.0",
    "@capacitor/device": "^1.0.0",
    "@capacitor/haptics": "^1.0.0",
    "@capacitor/ios": "^3.2.0",
    "@capacitor/keyboard": "^1.0.0"
  }
}
```

```java
package com.example.app;

import com.getcapacitor.BridgeActivity;

public class MainActivity extends BridgeActivity {}

```

Does `MainActivity.java` need to get modified manually in order for plugin loading to take effect? Should this file already have all the included plugins my app had already installed?

**Comment by jcesarmobile** [comment]
Capacitor reads the capacitor.plugins.json file that should contain the installed plugins, and with the information it finds there, it register the plugins without adding any code to the MainActivity.java.

There are a few other things that can cause the "plugin not implemented"
https://capacitorjs.com/docs/android/troubleshooting#plugin-not-implemented

**Comment by johnborges** [comment]
It seems I didn't even need to touch `MainActivity.java`. The issue causing "Plugin not implemented", was a typo in my plugin name annotation and an incorrect path to my main java file.
